src/utils/formHelperFunctions.js

```diff
@@ -73,6 +79,59 @@ const formatAllStepFields = (steps, fieldValues) => {
     return formattedFields;
 };
 
+const generateRegexString = (rawPatternString) => {
+    let startIndex = 0;
+    let regexString = '';
+    let patternComparisionString = '';
+    let patternMaskString = '';
+    let deterministic = true;
+
+    while (rawPatternString.indexOf('{{ ', startIndex) >= 0) {
+        const start = rawPatternString.indexOf('{{ ', startIndex);
+        const end = rawPatternString.indexOf(' }}', startIndex);
+        const identifiedRegex = rawPatternString.slice(start + 3, end).trim();
+
+        regexString += escapeChars(rawPatternString.slice(startIndex, start));
+        patternComparisionString += rawPatternString.slice(startIndex, start);
+        patternMaskString += rawPatternString.slice(startIndex, start);
+
+        regexString += identifiedRegex;
+
+        const expandedStrings = expandN(identifiedRegex, 500);
```

Comment:
i thought we discussed just matching on nondeterministic regex characters like * and + rather than trying to approximate its deterministic nature by generating 500 different options?

Reply:
There are different scenarios where the non-deterministic nature can pop up. So I have relied on this solution. I will be running some performance checks after the feature review is complete to see if things can be made faster.